crates/virtio-queue/src/chain.rs

```diff
@@ -88,6 +88,28 @@ where
         }
     }
 
+    /// Return a new instance of Writer
+    pub fn writer(self, mem: &'a GuestMemoryMmap) -> Result<Writer<'a>, Error>
+    where
+        M::Target: GuestMemory + Sized,
+    {
+        let Ok(writer) = Writer::new(&mem, self) else {
+            return Err(Error::InvalidChain);
+        };
+        Ok(writer)
+    }
+
+    /// Return a new instance of Reader
+    pub fn reader(self, mem: &'a GuestMemoryMmap) -> Result<Reader<'a>, Error>
+    where
+        M::Target: GuestMemory + Sized,
+    {
+        let Ok(reader) = Reader::new(&mem, self) else {
+            return Err(Error::InvalidChain);
+        };
+        Ok(reader)
```

Comment:
This could be a `.map_err()`.

Reply:
Do you mean something like:
```diff
-        let Ok(reader) = Reader::new(&mem, self) else {
-            return Err(Error::InvalidChain);
-        };
-        Ok(reader)
+        let reader = Reader::new(&mem, self).map_err(|_| Error::InvalidChain);
+        reader
```

Reply:
yes, though you can skip the assignment to the reader and just return the expression directly.